lib/utils/net-utils.js

```diff
@@ -0,0 +1,43 @@
+'use strict'
+
+const Promise = require('bluebird')
+const net = require('net')
+
+const NetUtils = {
+  isPortAvailable (port, listenAddress) {
+    return new Promise((resolve, reject) => {
+      const server = net.createServer()
+
+      server.unref()
+      server.on('error', (err) => {
+        server.close()
+        if (err.code === 'EADDRINUSE' || err.code === 'EACCES') {
+          resolve(false)
+        } else {
+          reject(err)
+        }
+      })
+
+      server.listen(port, listenAddress, () => {
+        server.close(() => resolve(true))
+      })
+    })
+  },
+
+  getAvailablePort (port, listenAddress, maxAttempts = 10) {
+    if (!maxAttempts) {
```

Comment:
I believe you can omit this check since you've added `err.code === 'EADDRINUSE' || err.code === 'EACCES'` check. 10 attempts aren't enough if you have hundreds test suites and you would like to parallelize them on multi-core machine

Reply:
Alright, `RangeError` should save us here. I'm gonna remove this `maxAttempts` check.

Reply:
Fixed